### PR TITLE
Add arguments step and out in curve.getSpacedPoints() method

### PR DESCRIPTION
**Pull request by rexrainbow**
This PR (delete as applicable)

* Adds a new feature

Describe the changes below:

Take an optional array as the 3rd parameter in which to store the points results.

### CODE REVIEW

File: src/curves/Curve.js [hunk 1]
```diff
@@ -353,24 +353,37 @@ var Curve = new Class({
      * @method Phaser.Curves.Curve#getSpacedPoints
      * @since 3.0.0
      *
-     * @param {integer} [divisions] - [description]
+     * @param {integer} divisions - The number of evenly spaced points from the curve to return. If falsy, step param will be used to calculate the number of points.
+     * @param {number} step - Step between points. Used to calculate the number of points to return when divisions is falsy. Ignored if divisions is positive.
```
Comment (photonstorm) [inline src/curves/Curve.js]: stepRate - also, optional

File: src/curves/Curve.js [hunk 2]
```diff
@@ -353,24 +353,37 @@ var Curve = new Class({
      * @method Phaser.Curves.Curve#getSpacedPoints
      * @since 3.0.0
      *
-     * @param {integer} [divisions] - [description]
+     * @param {integer} divisions - The number of evenly spaced points from the curve to return. If falsy, step param will be used to calculate the number of points.
```
Comment (photonstorm) [inline src/curves/Curve.js]: Optional

File: src/curves/Curve.js (hunk 1 above)
Comment (rexrainbow) [reply src/curves/Curve.js]: Fixed

File: src/curves/Curve.js (hunk 2 above)
Comment (rexrainbow) [reply src/curves/Curve.js]: Fixed

**Comment by photonstorm** [comment]
👍

**Pull request merged**